javascript/ql/src/Statements/UseOfReturnlessFunction.ql

```diff
@@ -82,8 +82,19 @@ predicate alwaysThrows(Function f) {
   )
 }
 
+/**
+ * Holds if the last statement in the function is flagged by the js/useless-expression query.
+ */
+predicate alwaysHasNoEffect(Function f) {
+  exists(ReachableBasicBlock entry, DataFlow::Node noEffect |
+    entry = f.getEntryBB() and
+    hasNoEffect(noEffect.asExpr()) and
+    entry.dominates(noEffect.getBasicBlock())
```

Comment:
I don't understand this condition. Doesn't the entry node dominate every basic block?

Reply:
Yes, you're right.  
I was a little quick in copy-pasting from `alwaysThrows` without thinking about what the predicate actually did.